modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -146,7 +146,19 @@ class PublishDir {
         final resolved = value instanceof Closure ? value.call() : value
         if( resolved instanceof String || resolved instanceof GString )
             nullPathWarn = checkNull(resolved.toString())
-        this.path = FileHelper.toCanonicalPath(resolved)
+        if( session?.cidEnabled ){
+            final resolvedPath = FileHelper.toPath(resolved)
+            if (resolvedPath.isAbsolute()){
+                log.warn("CID store is enabled but 'publishDir' is set to an absolute path ($resolvedPath). Outputs in this path will not published in the CID store")
+                this.path = FileHelper.toCanonicalPath(resolved)
+            }
+            else{
+                this.path = session.outputDir.resolve(resolvedPath)
+            }
+        }
+        else {
+            this.path = FileHelper.toCanonicalPath(resolved)
+        }
```

Comment:
Is this really needed? The CidObserver is already using the source path to determine the  origin of the file. 

https://github.com/nextflow-io/nextflow/pull/5787/files#diff-fe1fbb6658e26a22fa620333e674faabe3fbd0388f68cf78eda2dd50aea216beR167-R180

Reply:
Removed. Not needed in the last version